added pull-kubernetes-node-e2e-kubetest2 job

Namanl2001 opened the pull request:
Signed-off-by: Namanl2001 <namanlakhwani@gmail.com>

This PR is part of the CI migration from kubetest to kubetest2
https://github.com/kubernetes/enhancements/tree/master/keps/sig-testing/2464-kubetest2-ci-migration

Added new `pull-kubernetes-node-e2e-kubetest2` job

Required flags for this job in kubetest2 node tester(which were not present) are added via https://github.com/kubernetes-sigs/kubetest2/pull/166

xref: https://github.com/kubernetes-sigs/kubetest2/issues/164

Namanl2001 commented:
cc: @dims @amwat

dims commented:
/ok-to-test

Namanl2001 commented:
/retest

SergeyKanzhelev commented:
@Namanl2001 can you please provide more details on why this is needed

SergeyKanzhelev commented:
/assign @bobbypage 

to get more context on it and understand how it affects us

amwat commented:
This is part of the CI migration from kubetest to kubetest2 
https://github.com/kubernetes/enhancements/tree/master/keps/sig-testing/2464-kubetest2-ci-migration

amwat commented on config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:
```diff
@@ -86,6 +86,59 @@ presubmits:
+            memory: 6Gi
+        command:
+        - runner.sh
+        - /workspace/scenarios/kubernetes_e2e.py
```
we can drop the scenario arg

amwat commented on config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:
```diff
@@ -86,6 +86,59 @@ presubmits:
+      testgrid-num-failures-to-alert: "10"
+    spec:
+      containers:
+      - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20210915-5dbaf53458-master
```
let's use `...-experimental` instead of `...-master`
those will have the kubetest2 binaries.

amwat commented on config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:
```diff
@@ -86,6 +86,59 @@ presubmits:
+        - noop
+        - --build
+        - --up
+        - --down
```
We can probably drop all of the `--build` `--up` `--down` since they don't actually do anything.

amwat commented on config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:
```diff
@@ -86,6 +86,59 @@ presubmits:
+      timeout: 65m
+    labels:
+      preset-service-account: "true"
+      preset-dind-enabled: "true"
+      preset-kind-volume-mounts: "true"
```
let's remove these two, from first glance they seem to be redundant in the old job as well since usually `preset-dind-enabled` is associated with `securityContext: privileged` and are required if we are running docker commands in the job.

we can add them later again if we run into issues.

amwat commented on config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:
```diff
@@ -86,6 +86,59 @@ presubmits:
+        - --focus-regex='\[NodeConformance\]'
+        - --skip-regex='\[Flaky\]|\[Slow\]|\[Serial\]'
+        - --test-args='--kubelet-flags="--cgroups-per-qos=true --cgroup-root=/"'
+        - --image-config-file='/home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/image-config.yaml'
```
for now this is good to start with, once this job becomes stable let's add functionality in the node tester to be able to read files directly by specifying a raw url e.g. https://raw.githubusercontent.com/kubernetes/test-infra/master/jobs/e2e_node/image-config.yaml

spiffxp replied:
I generally don't like seeing prow's paths hardcoded in job configs, but this isn't a blocking suggestion
```suggestion
        - --image-config-file='../test-infra/jobs/e2e_node/image-config.yaml'
```

amwat commented on config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:
```diff
@@ -86,6 +86,59 @@ presubmits:
+    - master
+    decorate: true
+    path_alias: k8s.io/kubernetes
+    extra_refs:
+    - org: kubernetes
+      repo: test-infra
+      base_ref: master
+      path_alias: k8s.io/test-infra
```
we can drop the need to check out the test-infra repo after we add the raw url functionality mentioned in the other comment.

let's start with whatever we have for now.

Namanl2001 commented:
Thanks @amwat for a detailed review. Addressed all the comments.

We also need to add this job under some dashboard tab, so should I add this job under [this](https://github.com/kubernetes/test-infra/blob/22987c04fbfb97cf91b11431336c0ed1bea429d9/config/testgrids/kubernetes/sig-node/config.yaml#L34) tab?

amwat commented:
let's use  the `presubmits-kubernetes-nonblocking` tab: https://testgrid.k8s.io/presubmits-kubernetes-nonblocking

amwat commented on config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:
```diff
@@ -86,6 +86,53 @@ presubmits:
+      preset-service-account: "true"
+      preset-k8s-ssh: "true"
+    annotations:
+      fork-per-release: "true"
+      testgrid-alert-stale-results-hours: "24"
+      testgrid-create-test-group: "true"
+      testgrid-num-failures-to-alert: "10"
```
I think we can get rid of all of these. these annotation help enable alerting and make copies of the job on every branch. currently we only want it against master.

amwat commented on config/testgrids/kubernetes/presubmits/config.yaml:
```diff
@@ -147,6 +147,8 @@ dashboards:
     base_options: width=10
   - name: pull-kubernetes-node-e2e
     test_group_name: pull-kubernetes-node-e2e
+  - name: pull-kubernetes-node-e2e-kubetest2
```
I'm not sure if these are used, but looking at the other entries seems like this should also work.

In some other jobs I've seen annotations directly on the job e.g. https://github.com/kubernetes/test-infra/blob/3c87dfedd57e2dcad764a0d7fbb4343c7ca02a24/config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml#L103-L147

```
annotations:
      testgrid-dashboards: presubmits-kubernetes-nonblocking
      testgrid-tab-name: pull-kubernetes-node-e2e-kubetest2
```

Namanl2001 replied:
Yes, I think both should work. One advantage what I can see of adding in `config.yaml` is that we can also configure the width of test status tab.

amwat commented:
Changes look good!
/lgtm

cc @spiffxp for approval

Although the experimental image doesn't seem to automatically pick up changes

```
docker run --rm -it --entrypoint=kubetest2 gcr.io/k8s-staging-test-infra/kubekins-e2e:v20210915-5dbaf53458-experimental --version
kubetest2 version 4a034357d97563ff8726b716bd611b4292dbea0f
```

let me see if I can configure it to be a periodic instead of a postsubmit and/or need to add another script to fetch kubetest2 latest binaries at runtime.

spiffxp commented on config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:
```diff
@@ -86,6 +86,53 @@ presubmits:
+      testgrid-num-failures-to-alert: "10"
+    spec:
+      containers:
+      - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20210915-5dbaf53458-experimental
```
Probably needs a comment.  Why experimental / why not master?

spiffxp commented on config/testgrids/kubernetes/presubmits/config.yaml:
```diff
@@ -147,6 +147,8 @@ dashboards:
   - name: pull-kubernetes-node-e2e
     test_group_name: pull-kubernetes-node-e2e
+  - name: pull-kubernetes-node-e2e-kubetest2
+    test_group_name: pull-kubernetes-node-e2e-kubetest2
```
May as well go ahead and add this if you're adding it here to be able to do so
```suggestion
    test_group_name: pull-kubernetes-node-e2e-kubetest2
    base_options: width=10
```

spiffxp reviewed (changes requested):
A few nits but otherwise LGTM

spiffxp reviewed (approved):
/approve
/lgtm

Namanl2001 commented on config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:
```diff
@@ -86,6 +86,55 @@ presubmits:
+        - noop
+        - --test=node
+        - --
+        - --zone=us-west1-b
```
oops! Sorry. Added a [commit](https://github.com/kubernetes/test-infra/pull/23753/commits/24a77787f0035374bab39ccd1dee1e2b7815e08f) here.
